resources/2024-06-resource-extensibility-feature-spec.md

```diff
@@ -0,0 +1,452 @@
+            }
+        }
+
+        ```
+
+    1. VSCode Type spec extension warns Deb on the errors and warnings as he types
+     
+        ![alt text](2024-06-resource-extensbility/errors.png)
+
+    1. Deb uses intellisense to autocomplete the schema and properties
+
+        ![alt text](2024-06-resource-extensbility/intellisense.png)
+
+    1. Deb chooses a version for the schema
+
+        ```
+        import "@typespec/http";
+        import "@typespec/rest";
+        import "@typespec/openapi3";
+        import "@typespec/versioning";
+
+        using TypeSpec.Http;
+        using TypeSpec.Rest;
+        using TypeSpec.Versioning;
+
+        @service({title: "Contoso"})
+        @server("http://contoso.plaid.com", "endpoint for plaid service")
+        @versioned(Versions)
+        namespace Contoso;
+
+        @route("/planes/radius/Contoso.Messaging")
+        namespace plaid {
+            model plaid {
+                name:string;
+                description:string;
+                queue:string;
+            }
+        }
+
+        enum Versions {
+            v1,
+        }
+        ```
+   1. Deb registers the schema in Radius
+    
+        ```bash
+        rad resource-provider register contoso --template-kind typespec -e myenv -v v1
+        ```
+        Radius complies the schema and registers the custom resource type Plaid in UCP
+
+    1. Deb lists the resources types he has in his environment
+
+        ```bash
+        rad resource-provider list -e myenv
+        ```
+        Radius lists the custom resource types that Deb has registered in his environment
+
+2. Amy is a system integrator who is familiar with OpenAPI schema format and wants to author a custom resource PostgreSQL in Radius
+
+    1. Amy uses the radCLI to scaffold a yaml template definition
+
+        ```bash
+        rad resource-provider init plaid --template-kind yaml
+        ```
+        A sample yaml file is created in the application folder.
+
+    1. Amy authors the schema for plaid in typespec format
+        
+        ```
+        namespace: 'Applications.Datastores'
+        apiVersion: 'v1'
+        scope: '/planes/radius/Applications.Datastores'
+        resourceTypes:
+            'Applications.Datastores/postgreSQL':
+                name: 'Name of the resource'
+                description: 'PostgreSQL' 
+                properties: 
+                required: ['host', 'env', 'secrets']
+                host:
+                    type: 'string'
+                    description: 'Hostname'
+                env:
+                    type: 'string'
+                    description: 'Radius environment'
+                app:
+                    type: 'string'
+                    description: 'Radius application'
+                secrets:
+                    connectionstring:
+                    type: 'string'
+                    description: 'Connection string for PostgreSQL'
+        ```
+
+    1. Amy adds a version for the schema
```

Comment:
Does the yaml ultimately end up being compiled into the same format as the output as the TSP compiler? Or are we going to support two formats?

Reply:
I think we shd merge to the open API format typespec compiles to. Ultimately we want to be users of our own solution and refactor the Radius built in types to use UDT